channeldb/mp_payment.go

```diff
@@ -80,7 +80,7 @@ type MPPayment struct {
 
 	// HTLCs holds the information about individual HTLCs that we send in
 	// order to make the payment.
-	HTLCs []HTLCAttempt
+	HTLCs []*HTLCAttempt
```

Comment:
@halseth do you still know why you made this change originally?

Reply:
It wasn't done as an optimization, more as a style consistency change, since most other structs in the package are handled as pointer types (and because it is nice to just return `nil` in case of errors). I have no strong objection to reverting the change.

Reply:
Reverted to slice of values